tools/patch_charm_modules.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Applies any patches that are present in $3/tools/charm_module_patches/ to
+# their corresponding .def.h and decl.h files in the charm module outputs.
+# This is used to override default charm behavior with custom versions or to
+# patch bugs
+
+if [ "$#" -ne 4 ]; then
+    echo "Usage: patch_charm_modules.sh module_name path_to_current_source_dir\
+ path_to_source_root_dir path_to_current_build_dir"
+    exit
+fi
+
+PATCH_PREFIX=`echo "$2" | sed -En 's@'$3'@'$3'/tools/charm_module_patches@p'`
```

Comment:
`PATCH_PREFIX=$3/tools/charm_module_patches${2#"$3"}`
In words, that's: root directory followed by "/tools/charm_module_patches" followed by (current directory with root directory removed from the beginning).